metrics/cli/test/cli/command/update_command_test.dart

```diff
@@ -0,0 +1,189 @@
+
+void main() {
+  group("UpdateCommand", () {
+    const configPath = 'configPath';
+
+    final updaterFactory = _UpdaterFactoryMock();
+    final updateConfigFactory = _UpdateConfigFactoryMock();
+    final updater = _UpdaterMock();
+    final updateCommand = _UpdateCommandStub(
+      updaterFactory,
+      updateConfigFactory,
+      configPath: configPath,
+    );
+
+    PostExpectation<Updater> whenCreateUpdater() {
+      return when(updaterFactory.create());
+    }
+
+    tearDown(() {
+      reset(updaterFactory);
+      reset(updateConfigFactory);
+      reset(updater);
+    });
+
+    test(
+      "throws an ArgumentError if the given updater factory is null",
+      () {
+        expect(
+          () => DoctorCommand(null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".name equals to the 'update'",
+      () {
+        expect(updateCommand.name, equals('update'));
+      },
+    );
+
+    test(
+      ".description is not empty",
+      () {
+        final description = updateCommand.description;
+
+        expect(description, isNotEmpty);
+      },
+    );
+
+    test(
+      "has the 'config-file' option",
+      () {
+        final argParser = updateCommand.argParser;
+
+        expect(argParser.options, contains('config-file'));
+      },
+    );
+
+    test(
+      "'config-file' option has a non-empty help",
+      () {
+        final argParser = updateCommand.argParser;
+
+        final option = argParser.options['config-file'];
+
+        expect(option.help, isNotEmpty);
+      },
+    );
+
+    test(
+      "'config-file' option has a non-empty value help",
+      () {
+        final argParser = updateCommand.argParser;
+
+        final option = argParser.options['config-file'];
+
+        expect(option.valueHelp, isNotEmpty);
+      },
+    );
+
+    test(
+      ".run() creates updater using the given updater factory",
+      () async {
+        whenCreateUpdater().thenReturn(updater);
+
+        await updateCommand.run();
+
+        verify(updaterFactory.create()).called(once);
+      },
+    );
+
+    test(
+      ".run() uses the update config factory to create the update config",
+      () async {
+        whenCreateUpdater().thenReturn(updater);
+
+        await updateCommand.run();
+
+        verify(updateConfigFactory.create(configPath)).called(once);
+      },
+    );
+
+    test(
+      ".run() uses the updater to update the deployed Metrics components",
+      () async {
+        final updateConfig = UpdateConfigMock();
+
+        whenCreateUpdater().thenReturn(updater);
+        when(updateConfigFactory.create(configPath)).thenReturn(updateConfig);
+
+        await updateCommand.run();
+
+        verify(updater.update(updateConfig)).called(once);
+      },
+    );
+  });
+}
+
+/// A stub class for an [UpdateCommand] class providing test implementation
+/// for methods.
+class _UpdateCommandStub extends UpdateCommand {
```

Comment:
✅ 
I believe we should be careful mocking or stubbing the class under tests. This may violate encapsulation and provide side effects in the future. To solve this, we can introduce another class that would work with the given arg results. Or we can call the command under test from the above using the `run(args)` method of the runner. 

This is an interesting discussion about this problem https://softwareengineering.stackexchange.com/questions/107361/is-it-okay-to-fake-part-of-the-class-under-test

I suppose in our case splitting an object would work perfectly.